src/lua_console.cpp

```diff
@@ -0,0 +1,82 @@
+#include "lua_console.h"
+
+#include "catalua.h"
+#include "catacharset.h"
+
+lua_console::lua_console() : cWin( newwin( lines, width, 0, 0 ) ), iWin( newwin( 1, width,lines, 0 ) )
+{
+}
+
+void lua_console::dispose()
+{
+    werase( cWin );
+    werase( iWin );
+    delwin( cWin );
+    delwin( iWin );
+}
+
+std::string lua_console::get_input()
+{
+    long key = 0;
+    int pos = -1;
+    return string_input_win( iWin, "", width, 0, 0, width, true, key, pos, "LUA", 0, lines, true, false, callbacks );
+}
+
+void lua_console::draw()
+{
+    werase( cWin );
+
+    // Some juggling to make sure text is aligned with the bottom of the console.
+    int stack_size = text_stack.size() - scroll;
```

Comment:
> I like the idea, but putting all the text into one string would break the scrolling functionality. 

(Sorry, I forgot about this case.) Use `fold_and_print_from` ( it takes another parameters: the index of the first line (of the wrapped text) to be printed. Changing this number makes the text scroll. `draw_item_info` uses it that way.